q2_composition/tests/test_ancom.py

```diff
@@ -155,6 +155,26 @@ def test_ancom_no_tables(self):
             f = fh.read()
             self.assertTrue('No significant features found' in f)
 
+    def test_ancom_zero_division(self):
+        t = pd.DataFrame([[10, 0], [11, 0], [12, 0], [13, 0],
+                          [1000, 10], [1000, 10]],
+                         index=['S1', 'S2', 'S3', 'S4', 'S5', 'S6'],
+                         columns=['O1', 'O2'])
+        c = qiime2.CategoricalMetadataColumn(
+            pd.Series(['0', '0', '1', '1', '2', '2'], name='n',
+                      index=pd.Index(['S1', 'S2', 'S3', 'S4', 'S5', 'S6'],
+                                     name='id'))
+        )
+
+        ancom(output_dir=self.temp_dir.name, table=t+1, metadata=c,
+              transform_function='log')
```

Comment:
I'm curious why you incremented `t` here, rather than defining `t` with the desired values above.

Reply:
ANCOM can't operate on a table with any zeros in it, so we "impute" the table. A common imputation value is 1. As for why we added it in as a discrete step vs crafting the data that way from the get-go - no clue, I was just following what the other tests in this suite do (I know, I know, that's not a great excuse). I suppose it allows us to reason about the test data a bit more clearly, since we don't have to mentally subtract 1 from each value. I'm reaching here.